File: sliding-window/658-Find-K-Closest-Elements.go
func findClosestElements(arr []int, k int, x int) []int {
    currentDistance := 0
    for i := 0; i < k; i++ {
        currentDistance += abs(arr[i] - x)
    }

    left := 0
    minDistance := currentDistance
    for i := 1; i <= len(arr) - k; i++ {
        currentDistance = currentDistance - abs(arr[i - 1] - x) + abs(arr[i+k-1] - x)
        if currentDistance < minDistance {
            minDistance = currentDistance
            left = i
        }
    }
    
    return arr[left : left+k]
}

func abs (a int) int {
    if a < 0 {
        return -a
    }
    return a
}

File: sliding-window/658-Find-K-Closest-Elements.py
class Solution:
    def findClosestElements(self, arr: List[int], k: int, x: int) -> List[int]:
        minDistance, currentDistance = 0, 0
        for i in range(k):
            currentDistance += abs(arr[i] - x)
        
        startIndex = 0
        minDistance = currentDistance
        for i in range(1, len(arr) - k + 1):
            currentDistance += abs(arr[i+k-1] - x) - abs(arr[i-1] - x)
            if currentDistance < minDistance:
                minDistance = currentDistance
                startIndex = i
        
        return arr[startIndex: startIndex+k]